odoo/models.py

```diff
@@ -6373,69 +6373,49 @@ def flush_recordset(self, fnames=None):
-        # When a search on a field coming from a related occurs (the domain
-        # on discussion.important_emails field), make sure the related field
-        # is flushed
         if fnames is None:
             fields = self._fields.values()
         else:
             fields = [self._fields[fname] for fname in fnames]
 
-        model_fields = defaultdict(list)
-        for field in fields:
-            model_fields[field.model_name].append(field)
-            if field.related_field:
-                model_fields[field.related_field.model_name].append(field.related_field)
-
-        for model_name, fields_ in model_fields.items():
-            dirty_fields = self.env.cache.get_dirty_fields()
-            if any(field in dirty_fields for field in fields_):
-                # if any field is context-dependent, the values to flush should
-                # be found with a context where the context keys are all None
-                context_none = dict.fromkeys(
-                    key
-                    for field in fields_
-                    for key in self.pool.field_depends_context[field]
-                )
-                model = self.env(context=context_none)[model_name]
-                id_vals = defaultdict(dict)
-                for field in model._fields.values():
-                    ids = self.env.cache.clear_dirty_field(field)
-                    if not ids:
-                        continue
-                    records = model.browse(ids)
-                    values = list(self.env.cache.get_values(records, field))
-                    assert len(values) == len(records), \
-                        f"Could not find all values of {field} to flush them\n" \
-                        f"    Context: {self.env.context}\n" \
-                        f"    Cache: {self.env.cache!r}"
-                    for record, value in zip(records, values):
-                        if not field.translate:
-                            value = field.convert_to_write(value, record)
-                            value = field.convert_to_column(value, record)
-                        else:
-                            value = field._convert_from_cache_to_column(value)
-                        id_vals[record.id][field.name] = value
-                process(model, id_vals)
+        dirty_fields = self.env.cache.get_dirty_fields()
+        if not any(field in dirty_fields for field in fields):
+            return
 
-        # flush the inverse of one2many fields, too
-        for field in fields:
-            if field.type == 'one2many' and field.inverse_name:
-                self.env[field.comodel_name].flush_model([field.inverse_name])
+        # if any field is context-dependent, the values to flush should
+        # be found with a context where the context keys are all None
+        context_none = dict.fromkeys(
+            key
+            for field in fields
+            for key in self.pool.field_depends_context[field]
+        )
```

Comment:
There is a similar code here :
https://github.com/odoo/odoo/blob/5829efa1bfe7d5ac1bce8e9e85526131f2bd4a69/odoo/api.py#L1005

>  Indeed the values of context keys are taken from the context with context.get(key) and default to None.

Except for active_test but it shouldn't be a issue because `field.context` only exists for X2Many fields:
https://github.com/odoo/odoo/blob/5829efa1bfe7d5ac1bce8e9e85526131f2bd4a69/odoo/api.py#L792

Reply:
I created a separate commit for this one to test it on the runbot.